src/foam/core/AbstractArrayPropertyInfo.java

```diff
@@ -19,6 +19,15 @@
 public abstract class AbstractArrayPropertyInfo
   extends AbstractPropertyInfo {
 
+  public int compareValues(Object[] b1, Object[] b2) {
+    if (b1 == b2) return 0;
+    if (b2 == null) return 1;
+    if (b1 == null) return -1;
+    int h1 = b1.hashCode();
+    int h2 = b2.hashCode();
+    return h1 == h2 ? 0 : h1 > h2 ? 1 : -1;
```

Comment:
It isn't sufficient to compare hashCodes because different values can hash to the same value.

Reply:
Okay. I just copied this implementation over from AbstractFObjectArrayProperty. Not sure who originally wrote it. What would you suggest instead?

Reply:
@kgrgreer I ended up just removing the compareValues function because it wasn't being used anywhere to begin with